database/tables.sql

```diff
@@ -1,159 +1,230 @@
-);
-
--- EXAMEN_FISICO table
-CREATE TABLE examen_fisico (
-    id_examen_fisico SERIAL PRIMARY KEY,
-    descripcion TEXT NOT NULL,
-    frecuencia_respiratoria INTEGER,
-    temperatura DOUBLE PRECISION,
-    saturacion_oxigeno INTEGER,
-    glucometria INTEGER,
-    frecuencia_cardiaca INTEGER,
-    presion_arterial INTEGER
-);
-
--- VISITA table
-CREATE TABLE visita (
-    id_visita SERIAL PRIMARY KEY,
-    paciente VARCHAR(10) REFERENCES paciente (carnet),
-    motivo TEXT NOT NULL,
-    fecha TIMESTAMP NOT NULL DEFAULT NOW(),
-    diagnostico TEXT NOT NULL,
-    referencia TEXT,
-    examen_fisico INT REFERENCES examen_fisico (id_examen_fisico)
-);
-
--- TRATAMIENTO_VISITA table
-CREATE TABLE tratamiento_visita (
-    visita INT REFERENCES visita (id_visita),
-    tratamiento INT REFERENCES tratamiento (id_tratamiento),
-    detalle_farmacologia INT REFERENCES detalle_farmacologia (
-        id_detalle_farmacologia
-    )
-);
-
--- FORMULARIO table
-CREATE TABLE formulario (
-    id_formulario SERIAL PRIMARY KEY,
-    edicion VARCHAR(16) NOT NULL,
-    fecha TIMESTAMP NOT NULL
-);
-
--- RESPUESTA_FORMULARIO table
-CREATE TABLE respuesta_formulario (
-    formulario INT REFERENCES formulario (id_formulario) NOT NULL,
-    paciente VARCHAR(10) REFERENCES paciente (carnet) NOT NULL,
-    fecha TIMESTAMP NOT NULL DEFAULT NOW()
-);
-
--- PRIVILEGIO table
-CREATE TABLE privilegio (
-    id_privilegio SERIAL PRIMARY KEY,
-    privilegio VARCHAR(255) NOT NULL
-);
-
--- PRIVILEGIO_USUARIO table
-CREATE TABLE privilegio_usuario (
-    privilegio INT REFERENCES privilegio (id_privilegio) NOT NULL,
-    username VARCHAR(16) REFERENCES usuario (username) NOT NULL
+CREATE TABLE USUARIO (
+    EMAIL VARCHAR(50) PRIMARY KEY,
+    TIPO VARCHAR(20)
+);
+
+CREATE TABLE SEGURO (
+    ID SERIAL PRIMARY KEY,
+    NOMBRE VARCHAR(50),
+    ESTADO_ACTIVO BOOLEAN
+);
+
+CREATE TABLE PACIENTE (
+    ID SERIAL PRIMARY KEY,
+    CUI VARCHAR(24),
+    CORREO VARCHAR(50),
+    ES_MUJER BOOLEAN,
+    NOMBRES VARCHAR(24),
+    APELLIDOS VARCHAR(24),
+    NOMBRE_CONTACTO1 VARCHAR(50),
+    PARENTESCO_CONTACTO1 VARCHAR(24),
+    TELEFONO_CONTACTO1 VARCHAR(20),
+    NOMBRE_CONTACTO2 VARCHAR(50),
+    PARENTESCO_CONTACTO2 VARCHAR(24),
+    TELEFONO_CONTACTO2 VARCHAR(20),
+    TIPO_SANGRE VARCHAR(3),
+    DIRECCION VARCHAR(100),
+    ID_SEGURO INTEGER,
+    FECHA_NACIMIENTO DATE,
+    TELEFONO VARCHAR(20),
+    FOREIGN KEY (ID_SEGURO) REFERENCES SEGURO (ID)
+);
+
+CREATE TABLE ESTUDIANTE (
+    CARNET VARCHAR(20),
+    CARRERA VARCHAR(50),
+    ID_PACIENTE INTEGER NOT NULL,
+    PRIMARY KEY (CARNET),
+    FOREIGN KEY (ID_PACIENTE) REFERENCES PACIENTE (ID)
+);
+
+CREATE TABLE COLABORADOR (
+    CODIGO VARCHAR(20) PRIMARY KEY,
+    AREA VARCHAR(50),
+    ID_PACIENTE INTEGER NOT NULL,
+    FOREIGN KEY (ID_PACIENTE) REFERENCES PACIENTE (ID)
+);
+
+CREATE TABLE CONSULTA (
+    ID SERIAL PRIMARY KEY,
+    ID_PACIENTE INTEGER NOT NULL,
+    FECHA TIMESTAMP,
+    MOTIVO VARCHAR(100),
```

Comment:
Preferiría si el motivo de la consulta fuera un TEXT y también debería de ser NOT NULL porque no tiene sentido que ingresen una consulta sin haber ingresado un motivo primero.